test/metabase/models/field_test.clj

```diff
@@ -4,82 +4,3 @@
-(expect false (field-should-have-field-values? {:base_type       :type/Text
-                                                :special_type    :type/Latitude
-                                                :visibility_type :normal}))
-(expect false (field-should-have-field-values? {:base_type       :type/Text
-                                                :special_type    :type/Number
-                                                :visibility_type :normal}))
-(expect false (field-should-have-field-values? {:base_type       :type/Text
-                                                :special_type    :type/UNIXTimestampMilliseconds
-                                                :visibility_type :normal}))
-;; boolean fields + category/city/state/country fields are g2g
-(expect true (field-should-have-field-values? {:base_type       :type/Boolean
-                                               :special_type    :type/Number
-                                               :visibility_type :normal}))
-(expect true (field-should-have-field-values? {:base_type       :type/Text
-                                               :special_type    :type/Category
-                                               :visibility_type :normal}))
-(expect true (field-should-have-field-values? {:base_type       :type/Text
-                                               :special_type    :type/City
-                                               :visibility_type :normal}))
-(expect true (field-should-have-field-values? {:base_type       :type/Text
-                                               :special_type    :type/State
-                                               :visibility_type :normal}))
-(expect true (field-should-have-field-values? {:base_type       :type/Text
-                                               :special_type    :type/Country
-                                               :visibility_type :normal}))
-(expect true (field-should-have-field-values? {:base_type       :type/Text
-                                               :special_type    :type/Name
-                                               :visibility_type :normal}))
-
-
-;;; infer-field-special-type
-
-(tu/resolve-private-vars metabase.models.field infer-field-special-type)
-
-(expect nil            (infer-field-special-type nil       nil))
-(expect nil            (infer-field-special-type "id"      nil))
-(expect nil            (infer-field-special-type nil       :type/Integer))
-(expect :type/PK       (infer-field-special-type "id"      :type/Integer))
-;; other pattern matches based on type/regex (remember, base_type matters in matching!)
-(expect :type/Category (infer-field-special-type "rating"  :type/Integer))
-(expect nil            (infer-field-special-type "rating"  :type/Boolean))
-(expect :type/Country  (infer-field-special-type "country" :type/Text))
-(expect nil            (infer-field-special-type "country" :type/Integer))
```

Comment:
things you should never do:

delete all the code from a namespace and then leave the empty namespace there for no reason